server/src/test/java/org/elasticsearch/http/AbstractHttpServerTransportTests.java

```diff
@@ -1003,4 +1141,136 @@ public Collection<RestHeaderDefinition> getRestHeaders() {
             List.of()
         );
     }
+
+    private class TestHttpServerTransport extends AbstractHttpServerTransport {
+        public TestHttpChannel testHttpServerChannel = new TestHttpChannel(false);
+        public CountDownLatch gracePeriodCalled = new CountDownLatch(1);
+
+        TestHttpServerTransport(Settings settings) {
+            super(
+                Settings.builder().put(settings).put(SETTING_HTTP_CLIENT_STATS_ENABLED.getKey(), false).build(),
+                AbstractHttpServerTransportTests.this.networkService,
+                AbstractHttpServerTransportTests.this.recycler,
+                AbstractHttpServerTransportTests.this.threadPool,
+                xContentRegistry(),
+                new HttpServerTransport.Dispatcher() {
+                    @Override
+                    public void dispatchRequest(RestRequest request, RestChannel channel, ThreadContext threadContext) {
+                        channel.sendResponse(emptyResponse(RestStatus.OK));
```

Comment:
I think this is where some of my confusion above on idle terminology comes from. We should be able to block a request (hang here) to mimic an in flight request.

Reply:
`testStopForceClosesConnectionDuringRequest` which does that.